x/poe/contract/tgrade_valset.go

```diff
@@ -402,6 +401,31 @@ func (v ValsetContractAdapter) ListValidators(ctx sdk.Context, pagination *Pagin
 	return vals, cursor, nil
 }
 
+// IterateActiveValidators iterate through the active validator set.
+func (v ValsetContractAdapter) IterateActiveValidators(ctx sdk.Context, callback func(ValidatorInfo) bool) error {
+	var rsp ListActiveValidatorsResponse
+	cursor, err := v.doPageableQuery(ctx, ValsetQuery{ListActiveValidators: &ListValidatorsQuery{}}, &rsp)
+	if err != nil {
+		return err
+	}
+	for !cursor.Empty() {
+		for _, v := range rsp.Validators {
+			if callback(v) {
+				return nil
+			}
+		}
+		newCursor, err := v.doPageableQuery(ctx, ValsetQuery{ListActiveValidators: &ListValidatorsQuery{StartAfter: cursor.String()}}, &rsp)
+		if err != nil {
+			return err
+		}
+		if newCursor.Equal(cursor) {
```

Comment:
Huh? That is an odd check. When do they repeat?

(I thought the `cursor.Empty()` check would be enough)